nixos/modules/services/hardware/nvidia-container-toolkit-cdi-generator/cdi-generate.nix

```diff
@@ -11,6 +11,12 @@
       containerPath = "/usr/bin/nvidia-powerd"; }
     { hostPath = "${lib.getBin config.hardware.nvidia.package}/bin/nvidia-smi";
       containerPath = "/usr/bin/nvidia-smi"; }
+    # nvidia-docker 1.0 uses /usr/local/nvidia/lib{,64} on their LD_LIBRARY_PATH
+    #   e.g. https://gitlab.com/nvidia/container-images/cuda/-/blob/e3ff10eab3a1424fe394899df0e0f8ca5a410f0f/dist/12.3.1/ubi9/base/Dockerfile#L44
+    { hostPath = "${config.hardware.nvidia.package}/lib";
+      containerPath = "/usr/local/nvidia/lib"; }
+    { hostPath = "${config.hardware.nvidia.package}/lib";
+      containerPath = "/usr/local/nvidia/lib64"; }
```

Comment:
OK, so we're looking to institutionalize one more upstream hack that is broken by design and was never meant to work. Perhaps. Can we find a reference for the runtime wrapper actually populating these paths? Are we sure these paths aren't populated by the `yum install` from their `Dockerfile`?

Reply:
It's not about institutionalizing upstream hacks, in my opinion it's a matter of having a reasonable expectation of running container images that are already published in multiple places without having to worry about them as a user.

Why, as a NixOS user, do I have to worry about `podman run ollama/ollama` giving me an obscure error to learn that I need to tweak the `LD_LIBRARY_PATH` pointing to some `/nix/store` that changes over time? In my opinion that sounds miserable as a user experience.

Reply:
If upstream produces a broken image, the preference is to contact upstream and unbreak the image. This diff makes me suspect they might be building broken images (unless they do populate the path themselves in that `yum install`). Also it's extremely fishy that _you_ would need that `LD_LIBRARY_PATH` to work because it does not expose absolutely anything that you haven't exposed via `ld.so.cache` already.

TLDR: I'd like to understand what's going on a bit better

Reply:
> Can we find a reference for the runtime wrapper actually populating these paths? Are we sure these paths aren't populated by the yum install from their Dockerfile?

https://github.com/dalg24/nvidia-docker/blob/c68a91358e737e33aa949b45c8ad069ae2cac9b6/src/nvidia/volumes.go#L107